src/main/java/ch/njol/skript/conditions/CondFireworkHasDetonated.java

```diff
@@ -0,0 +1,69 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.Firework;
+import org.bukkit.entity.Projectile;
+
+@Name("Firework Is Detonated")
+@Description("Checks whether a firework has been detonated.")
+@Examples({
+	"set {_firework} to last launched firework",
+	"while {_firework} has not been detonated:",
+		"\tspawn tnt at {_firework}",
+		"\twait 10 ticks"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Spigot 1.19.4")
+public class CondFireworkHasDetonated extends PropertyCondition<Projectile> {
+
+	static {
+		if (Skript.methodExists(Firework.class, "isDetonated"))
+			Skript.registerCondition(CondFireworkHasDetonated.class, "%projectiles% (has|have) [:not] [been] detonated");
```

Comment:
A comment should be added to address why this isn't using the subclass register methods.
```suggestion
			// The negated pattern of PropertyCondition is not proper grammar for this syntax
			Skript.registerCondition(CondFireworkHasDetonated.class, "%projectiles% (has|have) [:not] [been] detonated");
```